File: bindings/node/examples/now_playing.js
// Live now-playing watcher — prints a line on every state change.
//
// Run with Spotify open and playing:
//   node bindings/node/examples/now_playing.js

'use strict';

const { SpotifyClient, Status } = require('..');

function fmt(ms) {
  const s = Math.max(ms | 0, 0) / 1000 | 0;
  return `${(s / 60) | 0}:${String(s % 60).padStart(2, '0')}`;
}

function main() {
  const client = new SpotifyClient();

  const render = (state) => {
    const pos = fmt(state.positionMs);
    const dur = fmt(state.durationMs);
    const label = state.statusName.toLowerCase();
    if (state.status === Status.UNKNOWN) {
      console.log(`[${label}] (Spotify not detected)`);
    } else {
      console.log(`[${label}] ${state.artist} — ${state.title}  [${pos} / ${dur}]`);
    }
  };

  client.on('stateChanged', render);

  const shutdown = () => {
    client.close();
    process.exit(0);
  };
  process.on('SIGINT', shutdown);
  process.on('SIGTERM', shutdown);

  client.start();
  render(client.latestState());  // initial baseline
}

main();

File: bindings/python/examples/now_playing.py
"""Live now-playing watcher — prints a line on every state change.

Run with Spotify open and playing:
    python bindings/python/examples/now_playing.py
"""

from __future__ import annotations

import signal
import threading
import time

from libspotifyctl import SpotifyClient, Status


def _fmt(ms: int) -> str:
    s = max(ms, 0) // 1000
    return f"{s // 60}:{s % 60:02d}"


def main() -> int:
    stop = threading.Event()
    signal.signal(signal.SIGINT, lambda *_: stop.set())

    def on_change(state) -> None:
        pos = _fmt(state.position_ms)
        dur = _fmt(state.duration_ms)
        label = state.status.name.lower()
        if state.status == Status.UNKNOWN:
            print(f"[{label}] (Spotify not detected)")
        else:
            print(f"[{label}] {state.artist} — {state.title}  [{pos} / {dur}]")

    with SpotifyClient() as client:
        client.on_state_changed(on_change)
        on_change(client.latest_state())  # print an initial baseline

        while not stop.is_set():
            stop.wait(timeout=0.25)

    return 0


if __name__ == "__main__":
    raise SystemExit(main())
